### ISSUE TITLE
How to send request from android client or ios , any sample?

### ISSUE THREAD

**Issue by surig** [google/shaka-player#789]
- What version of Shaka Player are you using?

  - Can you reproduce the issue with our latest release version?

  - Can you reproduce the issue with the latest code from `master`?

- Are you using the demo app or your own custom app?

  - If custom app, can you reproduce the issue using our demo app?

- What browser and OS are you using?

- What did you do?

  - What content did you load?

    - If standard demo asset, which one?

    - If custom asset, what manifest and license server URIs?

  - How did you interact with the content, if at all?

- What did you expect to happen?

- What actually happened?

**Comment by joeyparrish** [comment]
Hi @surig,

Can you please clarify what your question is?  You left our issue template completely blank, so all we have to go on is the issue title: "How to send request from android client or ios"

I don't understand what you mean by "send request" in this context.  Can you give us some more information on what you're trying to do?

**Comment by surig** [comment]
sorry, I mean regarding demo app i tried axinom drm content it is working fine from chrome shaka chrome browser.
I would like to check how to pass the parameters to receiver like license url , token with header message and content type ,If i want to send a sender request from android or IOS device with default android / IOS sender app.

**Comment by joeyparrish** [comment]
Support for generic sender apps was completed in #722 and first released in v2.1.0.  So this part should be working already.

To send parameters from your Shaka-based sender app to the Shaka-based receiver, use this API:

https://nightly-dot-shaka-player-demo.appspot.com/docs/api/shaka.cast.CastProxy.html#setAppData

For example, in the sender:

```js
player = new shaka.Player(videoElement);
castProxy = new shaka.cast.CastProxy(videoElement, player, receiverAppId);
castProxy.setAppData({
  anythingYouWant: { noWrongAnswers: true },
  thisIsAFreeFormObject: "containing whatever extra information you want to send.",
  authToken: "foo",
});

// Do not use setAppData for the license URL or manifest URL:
castProxy.getPlayer().configure({
  drm: {
    servers: {
      "com.widevine.alpha": licenseServerUri
    }
  }
});
castProxy.getPlayer().load(manifestUri);
```

Configuration and manifest URI will be proxied automatically to the receiver, but this mechanism can't transmit functions or callbacks, such as request filters.  Instead, pass your auth tokens via `setAppData()` and set up those callbacks in your receiver:

```js
player = new shaka.Player(video);
receiver = new shaka.cast.CastReceiver(video, player, onAppData);

function onAppData(appData) {
  console.log('App Data:', appData);
  player.getNetworkingEngine().clearAllRequestFilters();
  player.getNetworkingEngine().registerRequestFilter(function(type, request) {
    if (type != shaka.net.NetworkingEngine.RequestType.LICENSE) return;
    request.headers['MyAuthHeader'] = appData.authToken;
  });
}
```

Does this answer your question?

**Comment by joeyparrish** [comment]
Closing due to inactivity.  If you need any more help with this, please let us know and we can reopen.
